homeassistant/components/broadlink/__init__.py

```diff
@@ -21,11 +25,25 @@ async def async_setup(hass, config):
 
 async def async_setup_entry(hass, entry):
     """Set up a Broadlink device from a config entry."""
+    data = hass.data[DOMAIN]
+
+    if data.heartbeat is None:
+        data.heartbeat = BroadlinkHeartbeat(hass)
```

Comment:
Why shouldn't we just create a pinger per config entry? Using a single manager goes against the grain of our architecture which relies on config entries. It also makes the implementation more complicated. This could just be a simple function that we subscribe and unsubscribe with the event listener.

Reply:
In the previous implementation I was doing everything with a single call so as not to bomb the executor. Then I changed for some stupid reason during a clean up 😆

There are users with more than 10 devices and this method is called every 3 minutes, I think it makes sense to have a single for loop with all the calls running inside a single executor job. How about?

Reply:
How long does the ping take? I don't think we need to optimize it in that why unless we actually notice a problem.

Reply:
@MartinHjelmare I think we need to differentiate and not use the same measure for all integrations. There are 2 types of integrations: the type who only have a couple of devices/platforms, and for those we should do everything at platform level, but then we have another type f.x. zwave, modbus and this one, which are "carrier" (in lack of a better word) for potentially a lot of devices/platform, performance in HA, and the network load matters.  I think our architecture needs to allow for both cases (just think of a sensor device with 32 sensors, that should surely not be pinged 32 times).

Having a loop inside a single executor job, is the perfect middle way in my mind.

Reply:
I'm not against optimizing it if we have done the calculations and think it's a problem.